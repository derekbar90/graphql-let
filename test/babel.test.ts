import { deepStrictEqual } from 'assert';
import globby from 'globby';
import { join } from 'path';
import runner from '@babel/helper-transform-fixture-test-runner';
import { cleanup, readFile } from './__tools/file';

const cwd = join(__dirname, '__fixtures/babel');
const rel = (path: string) => join(cwd, path);

beforeAll(() => cleanup(cwd));

runner(cwd, 'gql', {}, { sourceType: 'unambiguous' });

test(`Generated files are okay`, async () => {
  const files = (await globby('fixtures/*/node_modules/**', { cwd })).sort();
  expect(files).toMatchSnapshot();

  for (const file of files)
    expect(await readFile(rel(file))).toMatchSnapshot(file);
});
