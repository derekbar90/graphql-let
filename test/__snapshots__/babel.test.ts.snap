// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generated files are okay: fixtures/*/node_modules/** 1`] = `
Array [
  "fixtures/basic/node_modules/@types/graphql-let/index.d.ts",
  "fixtures/basic/node_modules/@types/graphql-let/input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.d.ts",
  "fixtures/basic/node_modules/@types/graphql-let/store.json",
  "fixtures/basic/node_modules/graphql-let/__generated__/input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.tsx",
  "fixtures/tagged-template-call/node_modules/@types/graphql-let/index.d.ts",
  "fixtures/tagged-template-call/node_modules/@types/graphql-let/input-3daf5b4311465bd8566d77d548e135871cc11646.d.ts",
  "fixtures/tagged-template-call/node_modules/@types/graphql-let/store.json",
  "fixtures/tagged-template-call/node_modules/graphql-let/__generated__/input-3daf5b4311465bd8566d77d548e135871cc11646.tsx",
  "fixtures/with-fragments/node_modules/@types/graphql-let/index.d.ts",
  "fixtures/with-fragments/node_modules/@types/graphql-let/input-6500e855ce0d4683a1e6e953a93084b382d7a765.d.ts",
  "fixtures/with-fragments/node_modules/@types/graphql-let/store.json",
  "fixtures/with-fragments/node_modules/graphql-let/__generated__/input-6500e855ce0d4683a1e6e953a93084b382d7a765.tsx",
]
`;

exports[`Generated files are okay: fixtures/basic/node_modules/@types/graphql-let/index.d.ts 1`] = `
"import T1b047c726c8d0878f30e33cdadada1a79f3a6c62 from './input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.d.ts';
export default function gql(gql: \`query Viewer { viewer { name } }\`): T1b047c726c8d0878f30e33cdadada1a79f3a6c62.__AllExports;
"
`;

exports[`Generated files are okay: fixtures/basic/node_modules/@types/graphql-let/input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.d.ts 1`] = `
"/* 1b047c726c8d0878f30e33cdadada1a79f3a6c62
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export declare type Maybe<T> = T | null;
export declare type Exact<T extends {
  [key: string]: unknown;
}> = { [K in keyof T]: T[K] };
/** All built-in and custom scalars, mapped to their actual values */

export declare type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};
export declare type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};
export declare type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};
export declare type ViewerQueryVariables = Exact<{
  [key: string]: never;
}>;
export declare type ViewerQuery = ({
  __typename?: 'Query';
} & {
  viewer?: Maybe<({
    __typename?: 'User';
  } & Pick<User, 'name'>)>;
});
export declare const ViewerDocument: Apollo.DocumentNode;
/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */

export declare function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryResult<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryTuple<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export declare type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export declare type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;
export declare type __AllExports = {
  Maybe: Maybe;
  Exact: Exact;
  Scalars: Scalars;
  Query: Query;
  User: User;
  ViewerQueryVariables: ViewerQueryVariables;
  ViewerQuery: ViewerQuery;
  ViewerDocument: ViewerDocument;
  useViewerQuery: typeof useViewerQuery;
  useViewerLazyQuery: typeof useViewerLazyQuery;
  ViewerQueryHookResult: ViewerQueryHookResult;
  ViewerLazyQueryHookResult: ViewerLazyQueryHookResult;
  ViewerQueryResult: ViewerQueryResult;
};"
`;

exports[`Generated files are okay: fixtures/basic/node_modules/@types/graphql-let/store.json 1`] = `
"{
  \\"input.tsx\\": {
    \\"1b047c726c8d0878f30e33cdadada1a79f3a6c62\\": [
      \\"input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.d.ts\\",
      \\"query Viewer { viewer { name } }\\"
    ]
  }
}"
`;

exports[`Generated files are okay: fixtures/basic/node_modules/graphql-let/__generated__/input-1b047c726c8d0878f30e33cdadada1a79f3a6c62.tsx 1`] = `
"/* 1b047c726c8d0878f30e33cdadada1a79f3a6c62
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
const gql = Apollo.gql;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};

export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};

export type ViewerQueryVariables = Exact<{ [key: string]: never; }>;


export type ViewerQuery = (
  { __typename?: 'Query' }
  & { viewer?: Maybe<(
    { __typename?: 'User' }
    & Pick<User, 'name'>
  )> }
);


export const ViewerDocument = gql\`
    query Viewer {
  viewer {
    name
  }
}
    \`;

/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */
export function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
        return Apollo.useQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
      }
export function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
          return Apollo.useLazyQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
        }
export type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;"
`;

exports[`Generated files are okay: fixtures/tagged-template-call/node_modules/@types/graphql-let/index.d.ts 1`] = `
"import T3daf5b4311465bd8566d77d548e135871cc11646 from './input-3daf5b4311465bd8566d77d548e135871cc11646.d.ts';
export default function gql(gql: \`
    query Viewer {
        viewer {
            name
        }
    }
\`): T3daf5b4311465bd8566d77d548e135871cc11646.__AllExports;
"
`;

exports[`Generated files are okay: fixtures/tagged-template-call/node_modules/@types/graphql-let/input-3daf5b4311465bd8566d77d548e135871cc11646.d.ts 1`] = `
"/* 3daf5b4311465bd8566d77d548e135871cc11646
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export declare type Maybe<T> = T | null;
export declare type Exact<T extends {
  [key: string]: unknown;
}> = { [K in keyof T]: T[K] };
/** All built-in and custom scalars, mapped to their actual values */

export declare type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};
export declare type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};
export declare type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};
export declare type ViewerQueryVariables = Exact<{
  [key: string]: never;
}>;
export declare type ViewerQuery = ({
  __typename?: 'Query';
} & {
  viewer?: Maybe<({
    __typename?: 'User';
  } & Pick<User, 'name'>)>;
});
export declare const ViewerDocument: Apollo.DocumentNode;
/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */

export declare function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryResult<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryTuple<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export declare type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export declare type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;
export declare type __AllExports = {
  Maybe: Maybe;
  Exact: Exact;
  Scalars: Scalars;
  Query: Query;
  User: User;
  ViewerQueryVariables: ViewerQueryVariables;
  ViewerQuery: ViewerQuery;
  ViewerDocument: ViewerDocument;
  useViewerQuery: typeof useViewerQuery;
  useViewerLazyQuery: typeof useViewerLazyQuery;
  ViewerQueryHookResult: ViewerQueryHookResult;
  ViewerLazyQueryHookResult: ViewerLazyQueryHookResult;
  ViewerQueryResult: ViewerQueryResult;
};"
`;

exports[`Generated files are okay: fixtures/tagged-template-call/node_modules/@types/graphql-let/store.json 1`] = `
"{
  \\"input.tsx\\": {
    \\"3daf5b4311465bd8566d77d548e135871cc11646\\": [
      \\"input-3daf5b4311465bd8566d77d548e135871cc11646.d.ts\\",
      \\"\\\\n    query Viewer {\\\\n        viewer {\\\\n            name\\\\n        }\\\\n    }\\\\n\\"
    ]
  }
}"
`;

exports[`Generated files are okay: fixtures/tagged-template-call/node_modules/graphql-let/__generated__/input-3daf5b4311465bd8566d77d548e135871cc11646.tsx 1`] = `
"/* 3daf5b4311465bd8566d77d548e135871cc11646
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
const gql = Apollo.gql;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};

export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};

export type ViewerQueryVariables = Exact<{ [key: string]: never; }>;


export type ViewerQuery = (
  { __typename?: 'Query' }
  & { viewer?: Maybe<(
    { __typename?: 'User' }
    & Pick<User, 'name'>
  )> }
);


export const ViewerDocument = gql\`
    query Viewer {
  viewer {
    name
  }
}
    \`;

/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */
export function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
        return Apollo.useQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
      }
export function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
          return Apollo.useLazyQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
        }
export type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;"
`;

exports[`Generated files are okay: fixtures/with-fragments/node_modules/@types/graphql-let/index.d.ts 1`] = `
"import T6500e855ce0d4683a1e6e953a93084b382d7a765 from './input-6500e855ce0d4683a1e6e953a93084b382d7a765.d.ts';
export default function gql(gql: \`
  # import Partial from './partial.graphql'
  query Viewer {
    viewer {
      ...Partial
    }
  }
\`): T6500e855ce0d4683a1e6e953a93084b382d7a765.__AllExports;
"
`;

exports[`Generated files are okay: fixtures/with-fragments/node_modules/@types/graphql-let/input-6500e855ce0d4683a1e6e953a93084b382d7a765.d.ts 1`] = `
"/* 6500e855ce0d4683a1e6e953a93084b382d7a765
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export declare type Maybe<T> = T | null;
export declare type Exact<T extends {
  [key: string]: unknown;
}> = { [K in keyof T]: T[K] };
/** All built-in and custom scalars, mapped to their actual values */

export declare type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};
export declare type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};
export declare type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};
export declare type ViewerQueryVariables = Exact<{
  [key: string]: never;
}>;
export declare type ViewerQuery = ({
  __typename?: 'Query';
} & {
  viewer?: Maybe<({
    __typename?: 'User';
  } & PartialFragment)>;
});
export declare type PartialFragment = ({
  __typename?: 'User';
} & Pick<User, 'id' | 'name'>);
export declare const PartialFragmentDoc: Apollo.DocumentNode;
export declare const ViewerDocument: Apollo.DocumentNode;
/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */

export declare function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryResult<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>): Apollo.QueryTuple<ViewerQuery, Exact<{
  [key: string]: never;
}>>;
export declare type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export declare type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export declare type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;
export declare type __AllExports = {
  Maybe: Maybe;
  Exact: Exact;
  Scalars: Scalars;
  Query: Query;
  User: User;
  ViewerQueryVariables: ViewerQueryVariables;
  ViewerQuery: ViewerQuery;
  PartialFragment: PartialFragment;
  PartialFragmentDoc: PartialFragmentDoc;
  ViewerDocument: ViewerDocument;
  useViewerQuery: typeof useViewerQuery;
  useViewerLazyQuery: typeof useViewerLazyQuery;
  ViewerQueryHookResult: ViewerQueryHookResult;
  ViewerLazyQueryHookResult: ViewerLazyQueryHookResult;
  ViewerQueryResult: ViewerQueryResult;
};"
`;

exports[`Generated files are okay: fixtures/with-fragments/node_modules/@types/graphql-let/store.json 1`] = `
"{
  \\"input.tsx\\": {
    \\"6500e855ce0d4683a1e6e953a93084b382d7a765\\": [
      \\"input-6500e855ce0d4683a1e6e953a93084b382d7a765.d.ts\\",
      \\"\\\\n  # import Partial from './partial.graphql'\\\\n  query Viewer {\\\\n    viewer {\\\\n      ...Partial\\\\n    }\\\\n  }\\\\n\\"
    ]
  }
}"
`;

exports[`Generated files are okay: fixtures/with-fragments/node_modules/graphql-let/__generated__/input-6500e855ce0d4683a1e6e953a93084b382d7a765.tsx 1`] = `
"/* 6500e855ce0d4683a1e6e953a93084b382d7a765
 * This file is automatically generated by graphql-let. */

import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
const gql = Apollo.gql;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  viewer?: Maybe<User>;
};

export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  name: Scalars['String'];
  status: Scalars['String'];
};

export type ViewerQueryVariables = Exact<{ [key: string]: never; }>;


export type ViewerQuery = (
  { __typename?: 'Query' }
  & { viewer?: Maybe<(
    { __typename?: 'User' }
    & PartialFragment
  )> }
);

export type PartialFragment = (
  { __typename?: 'User' }
  & Pick<User, 'id' | 'name'>
);

export const PartialFragmentDoc = gql\`
    fragment Partial on User {
  id
  name
}
    \`;
export const ViewerDocument = gql\`
    query Viewer {
  viewer {
    ...Partial
  }
}
    \${PartialFragmentDoc}\`;

/**
 * __useViewerQuery__
 *
 * To run a query within a React component, call \`useViewerQuery\` and pass it any options that fit your needs.
 * When your component renders, \`useViewerQuery\` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useViewerQuery({
 *   variables: {
 *   },
 * });
 */
export function useViewerQuery(baseOptions?: Apollo.QueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
        return Apollo.useQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
      }
export function useViewerLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<ViewerQuery, ViewerQueryVariables>) {
          return Apollo.useLazyQuery<ViewerQuery, ViewerQueryVariables>(ViewerDocument, baseOptions);
        }
export type ViewerQueryHookResult = ReturnType<typeof useViewerQuery>;
export type ViewerLazyQueryHookResult = ReturnType<typeof useViewerLazyQuery>;
export type ViewerQueryResult = Apollo.QueryResult<ViewerQuery, ViewerQueryVariables>;"
`;
